tmt/template.py

```diff
@@ -0,0 +1,126 @@
+from typing import Any, Optional
+
+import tmt
+import tmt.utils
+from tmt.utils import Path, cached_property
+
+DEFAULT_CUSTOM_TEMPLATES_PATH = tmt.utils.CONFIG_PATH.expanduser() / 'templates'
+DEFAULT_PLAN_NAME = "/plans/default"
+INIT_TEMPLATES = ['mini', 'base', 'full']
+TEMPLATE_FILE_SUFFIX = '.j2'
+TEMPLATE_TYPES = ['default', 'story', 'plan', 'test', 'test_metadata']
+
+# TemplatesType is a dictionary of template types and their paths.
+# It follows the following structure: {template_type: {template_name: template_path}}
+# e.g. {'story': {'mini': Path('templates/story/mini.j2')}, ...}
+TemplatesType = dict[str, dict[str, Path]]
+
+
+def _combine(default: TemplatesType, custom: TemplatesType) -> TemplatesType:
+    """
+    Combines default templates and custom templates.
+    Custom templates have priority and potentially override default templates.
+    """
+    result: TemplatesType = {}
+    for key in default:
+        result[key] = {**default[key], **custom.get(key, {})}
+    for key in custom:
+        if key not in result:
+            result[key] = custom[key]
```

Comment:
Is the second loop needed? Aren't all keys from the `custom` dict already assigned during the first loop?

Reply:
You are right, the keys will always be assigned during the first loop. Removed in eca25386d95a1e4b9f879f3df89d871720787173.